File: 2022/javascript/day_2/day_2_2.js
const {readFileSync} = require("fs")

const input = readFileSync("./input.txt", "utf-8").replace(/\r/g, "")

// A => Rock
// B => Paper
// C => Scissors

// Rock => 1
// Paper => 2
// Scissors => 3

// X => Lose
// Y => Draw
// Z => Win

const games = input.split("\n")

const results = games.map(g => {
    const [opponent, outcome] = g.split(" ")
    let score = 0
    switch(outcome) {
        case "X": {
            score += opponent === "A" ? 3 : opponent === "B" ? 1 : 2;
            break;
        }
        case "Y": {
            score += 3
            score += opponent === "A" ? 1 : opponent === "B" ? 2 : 3;
            break;
        }
        case "Z": {
            score += 6
            score += opponent === "A" ? 2 : opponent === "B" ? 3 : 1;
            break;
        }
    }
    return score
})

const total = results.reduce((a, b) => a + b)

console.log(total)

File: 2022/cpp/day_2/day_2_2.cpp
#include "day_2.h"
#include <string>
#include <fstream>
#include <vector>
using namespace std;

// a - Rock
// b - Paper
// c - Scissors

// x - lose
// y - draw
// z - win

int day_2_2() {
    string line;

    ifstream MyReadFile("../day_2/input.txt");

    vector<string> game;

    int score = 0;

    while (getline (MyReadFile, line)) {
        const char opponentShape = line[0];
        const char winType = line[2];

        score += winType == 'Z' ? 6 : winType == 'Y' ? 3 : 0;

        if(opponentShape == 'A') {
            score += winType == 'Z' ? 2 : winType == 'Y' ? 1 : 3;
        } else if(opponentShape == 'B') {
            score += winType == 'Z' ? 3 : winType == 'Y' ? 2 : 1;
        } else if(opponentShape == 'C') {
            score += winType == 'Z' ? 1 : winType == 'Y' ? 3 : 2;
        }
    }

    printf("Final score: %i", score);

    return 0;
}